tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -67,6 +67,15 @@ static const std::string _register_names[11] = {
     "r10",
 };
 
+static const std::set<std::string> _callee_register_args = {
+    "r1",
+    "r2",
+    "r3",
+    "r4",
+    "r5",
+    "r10",
```

Comment:
nit: why do we need to pass stack pointer to the sub-program? The current sample program does not use this? Should we enhance the sample programs to test cases where the sub-program uses `r10` also?

Reply:
Yes it's part of the future PR I referred to in the PR description, for which https://github.com/vbpf/ebpf-verifier/pull/721 is the prerequisite.